### PR TITLE
Add tests to clientprocessorforjava

### PR DESCRIPTION
**Pull request by bennsimon**
Added the following tests for ClientProcessorForJava:

- `closeCase`
- `updateIdenitifier`
- `getGender`
- `getClientAttributes`

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java [hunk 1]
```diff
@@ -663,7 +658,7 @@ public void updateClientDetailsTable(Event event, Client client) {
             if (StringUtils.isNotBlank(gender)) {
                 map.put(GENDER, gender);
             }
-        } catch (Exception e) {
+        } catch (NullPointerException e) {
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java]: Is this the only the exception that can occur

File: opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java [hunk 2]
```diff
@@ -648,7 +643,7 @@ public void updateClientDetailsTable(Event event, Client client) {
                     attributes.put(key, value.toString());
                 }
             }
-        } catch (Exception e) {
+        } catch (NullPointerException e) {
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java]: Is this the only the exception that can occur

File: opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorForJavaTest.java [hunk 3]
```diff
@@ -92,4 +114,57 @@ public void testGetValuesStrShouldGetCorrectlyFormattedString() throws Exception
...
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        CommonRepository commonRepository = Mockito.mock(CommonRepository.class);
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", coreLibrary);
+        PowerMockito.when(coreLibrary.context()).thenReturn(opensrpContext);
+        PowerMockito.when(opensrpContext.commonrepository("child")).thenReturn(commonRepository);
+        assertTrue(clientProcessor.closeCase(new Client("1233-2"), Arrays.asList("child")));
+        Mockito.verify(commonRepository).closeCase(closeCaseArgumentCaptor.capture(), closeCaseArgumentCaptor.capture());
+        assertEquals("1233-2", closeCaseArgumentCaptor.getAllValues().get(0));
+        assertEquals("child", closeCaseArgumentCaptor.getAllValues().get(1));
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", null);
```
Comment (githengi) [inline opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorForJavaTest.java]: why is this required, unit tests should normally end with asserts or expected failures

File: opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java (hunk 2 above)
Comment (bennsimon) [reply opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java]: yeah i think so

File: opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java (hunk 1 above)
Comment (bennsimon) [reply opensrp-app/src/main/java/org/smartregister/sync/ClientProcessorForJava.java]: yeah i think so

File: opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorForJavaTest.java (hunk 3 above)
Comment (bennsimon) [reply opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorForJavaTest.java]: let add a teardown for this

**Pull request merged**